Zend/zend_API.c

```diff
@@ -2772,6 +2772,9 @@ ZEND_END_ARG_INFO()
 
 static zend_always_inline void zend_normalize_internal_type(zend_type *type) {
 	ZEND_ASSERT(!ZEND_TYPE_HAS_LITERAL_NAME(*type));
+	if (ZEND_TYPE_PURE_MASK(*type) != MAY_BE_ANY) {
```

Comment:
In light of https://github.com/php/php-src/pull/13159#pullrequestreview-1822996566, this should probably be wrapped in `#if DEBUG`.

Reply:
I mean, it's not super important. The other assert is potentially subject to the same. So I guess this is fine.